pkg/querier/codec/protobuf_codec.go

```diff
@@ -0,0 +1,158 @@
+package codec
+
+import (
+	"github.com/cortexproject/cortex/pkg/querier/tripperware"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/prometheus/prometheus/util/stats"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/web/api/v1"
+	"github.com/gogo/protobuf/proto"
+)
+type ProtobufCodec struct{}
+
+func (p ProtobufCodec) ContentType() v1.MIMEType {
+	return v1.MIMEType{Type: "application", SubType: "x-protobuf"}
+}
+
+func (p ProtobufCodec) CanEncode(resp *v1.Response) bool {
+	// Errors are parsed by default json codec
+	if resp.Error != "" || resp.Data == nil {
+		return false
+	}
+	return true
+}
+
+func (p ProtobufCodec) Encode(resp *v1.Response) ([]byte, error) {
+	prometheusInstantQueryResponse, err := createPrometheusInstantQueryResponse(resp)
+	if err != nil {
+		return []byte{}, err
+	}
+	b, err := proto.Marshal(prometheusInstantQueryResponse)
+	return b, err
+}
+
+func createPrometheusInstantQueryResponse(resp *v1.Response) (*tripperware.PrometheusResponse, error) {
+	var data = resp.Data.(*v1.QueryData)
+
+	var instantQueryResult tripperware.PrometheusQueryResult
```

Comment:
Let's not remove `instant` prefix

Reply:
I was actually planning to remove it since we are handling both instant and range queries the same way. Shouldn't we rename it to `createPrometheusQueryResponse()` to reflect that?

Reply:
Got it. Let's do that instead

Reply:
done